src/gocam/cli.py

```diff
@@ -476,5 +480,273 @@ def flatten_models(input_file, input_format, output_format, output_file, fields)
+        formats = list(formats)
+    
+    # Create output directories
+    output_paths = {}
+    if "networkx" in formats:
+        output_paths["networkx"] = output_networkx
+        os.makedirs(output_networkx, exist_ok=True)
+        click.echo(f"Created output directory for NetworkX: {output_networkx}", err=True)
+    
+    if "cx2" in formats:
+        output_paths["cx2"] = output_cx2
+        os.makedirs(output_cx2, exist_ok=True)
+        click.echo(f"Created output directory for CX2: {output_cx2}", err=True)
+    
+    # Create temporary directory for extraction
+    with tempfile.TemporaryDirectory() as temp_dir:
+        try:
+            # Download and extract the tarball
+            extracted_dir = download_and_extract_tarball(url, temp_dir)
+            
+            # Find all JSON model files
+            json_files = find_json_models(extracted_dir)
+            
+            if limit:
+                json_files = json_files[:limit]
+                click.echo(f"Limited processing to {len(json_files)} models", err=True)
+            
+            # Process each model
+            processed_count = 0
+            failed_count = 0
+            processed_model_ids = []
+            
+            for json_file in json_files:
+                model_filename = os.path.splitext(os.path.basename(json_file))[0]
+                logger.info(f"Processing model: {model_filename}")
+                
+                # Load the model
+                model = load_model_from_file(json_file)
+                if model is None:
+                    failed_count += 1
+                    continue
+                
+                # Track successfully processed model ID
+                processed_model_ids.append(model.id)
+                
+                # Translate to requested formats
+                if "networkx" in formats:
+                    translate_to_networkx(model, output_paths["networkx"], model_filename)
+                
+                if "cx2" in formats:
+                    translate_to_cx2(model, output_paths["cx2"], model_filename)
+                
+                processed_count += 1
+            
+            click.echo(f"Processing complete. Successfully processed: {processed_count}, Failed: {failed_count}", err=True)
+            
+            # Create tar.gz archives if requested
+            if archive:
+                def create_archive(directory_path: str, format_name: str):
+                    """Create a gzipped tar archive of a directory."""
+                    if not os.path.exists(directory_path) or not os.listdir(directory_path):
+                        click.echo(f"Skipping archive for {format_name}: directory is empty or doesn't exist", err=True)
+                        return
+                    
+                    # Create archive filename
+                    import datetime
+                    schema_version = __version__
+                    archive_name = f"gocam_{format_name}.tar.gz"
```

Comment:
If this is meant to be a public product, is there a reason not to use "go-cam" here? Also noting that a lot of our public products use "-" instead of "_".
That said, neither are a blocker for me in the grand scheme of things.

Reply:
went with `go-cam-networkx.tar.gz` and `go-cam-cx2.tar.gz`